File: 백준/Silver/11501. 주식/주식.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int t = in.nextInt();
        for (int i = 0; i < t; i++) {
            int n = in.nextInt();
            long[] arr = new long[n];
            for (int j = 0; j < n; j++) {
                arr[j] = in.nextInt();
            }

            long profit = 0;
            long maxPrice = 0;

            for (int j = arr.length - 1; j >= 0; j--) {
                if (arr[j] > maxPrice) {
                    maxPrice = arr[j];
                } else {
                    profit += maxPrice - arr[j];
                }
            }
            System.out.println(profit);
        }

    }
}

File: 백준/Silver/11501. 주식/주식.py
import sys

input = sys.stdin.readline

for _ in range(int(input())):
    n = int(input())
    arr = list(map(int, input().split()))

    profit = 0
    max_price = 0
    for i in range(len(arr)-1, -1, -1):
        if arr[i] > max_price:
            max_price = arr[i]
        else:
            profit += max_price - arr[i]

    print(profit)